uni/lib/controller/background_workers/notifications.dart

```diff
@@ -0,0 +1,150 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:logger/logger.dart';
+import 'package:tuple/tuple.dart';
+import 'package:uni/controller/background_workers/notifications/tuition_notification.dart';
+import 'package:uni/controller/local_storage/app_shared_preferences.dart';
+import 'package:uni/controller/local_storage/notification_timeout_storage.dart';
+import 'package:uni/controller/networking/network_router.dart';
+import 'package:uni/model/entities/session.dart';
+import 'package:workmanager/workmanager.dart';
+
+///
+/// Stores all notifications that will be checked and displayed in the background. 
+/// Add your custom notification here, because it will NOT be added at runtime.
+/// (due to background worker limitations).
+/// 
+Map<Type, Notification Function()> notificationMap = {
+  TuitionNotification:() => TuitionNotification(),
+};
+
+
+
+
+abstract class Notification{
+  
+  String uniqueID;
+  Duration timeout;
+
+  Notification(this.uniqueID, this.timeout);
+
+  Future<Tuple2<String, String>> buildNotificationContent(Session session);
+
+  Future<bool> checkConditionToDisplay(Session session);
+
+  void displayNotification(Tuple2<String, String> content, FlutterLocalNotificationsPlugin localNotificationsPlugin);
+
+  Future<void> displayNotificationIfPossible(Session session, FlutterLocalNotificationsPlugin localNotificationsPlugin) async{
+    final bool test = await checkConditionToDisplay(session);
+    if(test){
+      displayNotification(await buildNotificationContent(session), localNotificationsPlugin);
+    }
+  }
+}
+
+class NotificationManager{
```

Comment:
It makes sense that this is a singleton, however, I believe we should make this clearer: https://flutterbyexample.com/lesson/singletons

Reply:
Done